crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -832,6 +833,22 @@ fn string_type() -> impl NoirParser<UnresolvedType> {
         .map(UnresolvedType::String)
 }
 
+fn format_string_type(
+    type_parser: impl NoirParser<UnresolvedType>,
+) -> impl NoirParser<UnresolvedType> {
+    let fields = type_parser.separated_by(just(Token::Comma)).allow_trailing();
+
+    keyword(Keyword::FormatString)
+        .ignore_then(
+            type_expression()
+                .or_not()
+                .then_ignore(just(Token::Comma))
+                .then(parenthesized(fields))
+                .delimited_by(just(Token::Less), just(Token::Greater)),
```

Comment:
I think it's rather odd/unintuitive that we force parenthesis when parsing fmt string types. We should also allow `fmtstr<N, T>` for example. This is similar to a closure's environment. I think the syntax should be roughly "FmtStr < type_parser , type_parser >"

Reply:
Yeah you are correct it is odd. I will update this

Reply:
This has been updated. We can now handle these different types of `fmtstr` types:
```
fmtstr<14, (Field, Field)>
fmtstr<N, T>
fmtstr<N, (Field, T)>
```